src/core/State.cpp

```diff
@@ -47,25 +47,29 @@
 #include <QtXmlPatterns/QXmlQuery>
 
 State::State()
-    : version_( INVALID_FILE_VERSION )
+    : displayGroup_( new DisplayGroup( QSizeF( 0 , 0 )))
+    , version_( INVALID_FILE_VERSION )
+{}
+
+State::State( const DisplayGroupPtr displayGroup )
+    : displayGroup_( displayGroup )
+    , version_( INVALID_FILE_VERSION )
 {
 }
 
-State::State(const ContentWindowPtrs& contentWindows)
-    : contentWindows_( contentWindows )
-    , version_( INVALID_FILE_VERSION )
+ContentWindowPtrs& State::getContentWindows() const
 {
+    return displayGroup_->getContentWindows();
 }
 
-const ContentWindowPtrs& State::getContentWindows() const
+DisplayGroup& State::getDisplayGroup() const
```

Comment:
If you return a non-const reference to an internal object, this method should not be const even if the compiler lets you, because it would allow a user modify the internals of a const State...